app/code/core/Mage/Eav/Model/Config.php

```diff
@@ -186,8 +186,23 @@ protected function _loadEntityTypes()
         $this->_entityTypes = [];
         $this->_entityTypeByCode = [];
         $entityTypeCollection = Mage::getResourceModel('eav/entity_type_collection');
+
         /** @var Mage_Eav_Model_Entity_Type $entityType */
         foreach ($entityTypeCollection as $entityType) {
+            // Ensure eav entity type model class is defined, otherwise skip processing it.
+            // This check prevents leftover eav_entity_type entries from disabled/removed modules creating errors and
+            // is necessary because the entire EAV model is now loaded eagerly for performance optimization.
+            $entityModelClass = $entityType['entity_model'];
+            $fqEntityModelClass = Mage::getConfig()->getModelClassName($entityModelClass);
+            if (!class_exists($fqEntityModelClass)) {
+                if (Mage::getIsDeveloperMode()) {
+                    throw new Exception('failed loading of eav entity type because it does not exist: ' . $entityModelClass);
+                } else {
+                    Mage::log('skipped loading of eav entity type because it does not exist: ' . $entityModelClass);
+                }
+                continue;
+            }
+
```

Comment:
There should be a capital letter at the beginning of the exception messages to keep the rule we already have. The message that starts with a lowercase letter can appear as a phrase added to another message.